### PR TITLE
Add async label for waitMicros

### PR DESCRIPTION
**Pull request by livcheerful**
fixes https://github.com/microsoft/pxt-microbit/issues/3817

Wasn't sure what the difference between changing control.cpp vs shims.d.ts, so I added async to both! Let me know what the difference is and if I should remove it lol

### CODE REVIEW

**Review by jwunderl — APPROVED** [review]

File: libs/core/control.cpp [hunk 1]
```diff
@@ -273,7 +273,7 @@ namespace control {
     * Blocks the current fiber for the given microseconds
     * @param micros number of micro-seconds to wait. eg: 4
     */
-    //% help=control/wait-micros weight=29
+    //% help=control/wait-micros weight=29 async
```
Comment (pelikhan) [inline libs/core/control.cpp:276]: wait Micros is blocking.

**Review by pelikhan — CHANGES_REQUESTED** [review]
waitMicros is blocking by design.

**Comment by pelikhan** [comment]
The async label was used in TouchDevelop times; i'm not sure it's relevant anymore. Needs to be checked.

It is misleading to mark waitMicros as async as it is blocking on hardware. common-packages version should be fixed i guess. It does create a discreptancy in the way we simulate the code in the browser but there's not much we can do there.

Why the change?

**Comment by pelikhan** [comment]
To be clear: waitMicros is mean to be used with a **small** amount of micro to do real-time computation (like measuring time between 2 pin pulses). Anything larger than a few hundred microseconds, should use pause which has a minimum delay of 6ms.

waitMicros hogs the MCU, and thus will start fighting the BLE stack and so on.

**Review by pelikhan — APPROVED** [review]

**Pull request merged**